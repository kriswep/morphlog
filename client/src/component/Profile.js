import React from 'react';
import gql from 'graphql-tag';
import { graphql, compose } from 'react-apollo';

class Profile extends React.Component {
  state = {
    email: '',
    password: '',
  };

  dispatch = (e, v, x) => {
    if (e.target.dataset && e.target.dataset.state) {
      const newState = {};
      newState[e.target.dataset.state] = e.target.value;
      this.setState(newState);
    }
  };

  signup = e => {
    this.props
      .signupMutation({
        variables: {
          email: this.state.email,
          password: this.state.password,
        },
      })
      .then(({ data }) => {
        console.log('got data', data);
        localStorage.setItem('auth', data.signup.token);
      })
      .catch(error => {
        console.log('there was an error sending the query', error);
      });
  };

  signin = e => {
    this.props
      .signinMutation({
        variables: {
          email: this.state.email,
          password: this.state.password,
        },
      })
      .then(({ data }) => {
        console.log('got data', data);
        localStorage.setItem('auth', data.login.token);
      })
      .catch(error => {
        console.log('there was an error sending the query', error);
      });
  };

  render() {
    console.log(this.props.profileQuery);
    return (
      <div>
        <h2>Profile</h2>
        <input
          type="text"
          placeholder="email"
          data-state="email"
          onChange={this.dispatch}
        />
        <input
          type="password"
          placeholder="password"
          data-state="password"
          onChange={this.dispatch}
        />
        <button onClick={this.signup}>SignUp</button>
        <button onClick={this.signin}>SignIn</button>
        {!this.props.profileQuery.loading && (
          <section>
            <p>Name: {this.props.profileQuery.me.name}</p>
            <p>E-Mail: {this.props.profileQuery.me.email}</p>
          </section>
        )}
      </div>
    );
  }
}

const SIGNUP_MUTATION = gql`
  mutation s($email: String!, $password: String!) {
    signup(email: $email, password: $password, name: "demo3") {
      token
      user {
        id
        name
        email
      }
    }
  }
`;
const SIGNIN_MUTATION = gql`
  mutation signin($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        id
        name
        email
      }
    }
  }
`;
const PROFILE_QUERY = gql`
  query profile {
    me {
      id
      name
      email
    }
  }
`;

export default compose(
  graphql(PROFILE_QUERY, { name: 'profileQuery' }),
  graphql(SIGNUP_MUTATION, { name: 'signupMutation' }),
  graphql(SIGNIN_MUTATION, { name: 'signinMutation' }),
)(Profile);
