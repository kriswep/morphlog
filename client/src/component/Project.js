import React from 'react';
import { gql } from 'apollo-boost';
import { graphql, compose } from 'react-apollo';
import styled from 'styled-components';

import media from '../styles/media';
import Changes from './Changes';

const ContentContainer = styled.section`
  padding: 1rem 0.25rem 0.75rem;
  ${media.m`
    padding: 1rem 0.75rem 0.75rem;
  `};
`;

const initialState = {};

export class Project extends React.Component {
  state = initialState;

  // dispatch = (e, v, x) => {
  //   if (e.target.name && e.target.name) {
  //     const newState = {};
  //     newState[e.target.name] = e.target.value;
  //     this.setState(newState);
  //   }
  // };

  render() {
    const project = this.props.projectQuery.project;
    return (
      <ContentContainer>
        {project && (
          <div>
            <h2>{project.name}</h2>
            <Changes projectId={this.props.projectQuery.project.id} />
          </div>
        )}
      </ContentContainer>
    );
  }
}

const PROJECT_QUERY = gql`
  query project($id: ID!) {
    project(id: $id) {
      id
      name
    }
  }
`;

export default compose(
  graphql(PROJECT_QUERY, {
    name: 'projectQuery',
    options: props => ({
      variables: { id: props.projectId || props.match.params.projectId },
    }),
  }),
)(Project);
