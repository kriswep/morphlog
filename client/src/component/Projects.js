import React from 'react';
import gql from 'graphql-tag';
import { graphql, compose } from 'react-apollo';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { Route, Redirect } from 'react-router';

import media from '../styles/media';
import Project from './Project';
import Input from './Input';

const SidebarContainer = styled.section`
  grid-area: sidebar;
  padding: 0.25rem;
  ${media.m`
    padding 0.75rem;
  `};
`;

const ProjectsContainer = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0;
`;

const ProjectLink = styled(Link)`
  color: ${props => props.theme.darkShades};
`;

const initialState = {
  name: '',
};

const ProjectWrapper = props => {
  const projectId = props;
  return <Project {...props} />;
};

class Projects extends React.Component {
  state = initialState;

  dispatch = (e, v, x) => {
    if (e.target.name && e.target.name) {
      const newState = {};
      newState[e.target.name] = e.target.value;
      this.setState(newState);
    }
  };

  addProject = e => {
    this.props
      .addProjectMutation({
        variables: {
          name: this.state.name,
        },
        refetchQueries: [
          {
            query: PROJECTS_QUERY,
          },
        ],
      })
      .then(({ data }) => {
        console.log('got data', data);
      })
      .catch(error => {
        console.log('there was an error sending the query', error);
      })
      .finally(() => {
        this.setState(initialState);
      });
  };

  render() {
    const projects = this.props.projectsQuery.projects;

    let projectId = this.props.match.params.projectId;
    if (
      !this.props.match.params.projectId &&
      projects &&
      projects.length >= 0 &&
      projects[0].id
    ) {
      // no project open, but we have projects: render project comp
      projectId = projects[0].id;
    }
    return [
      projectId && <Project projectId={projectId} />,
      <SidebarContainer>
        <h2>Projects</h2>
        <Input
          name="name"
          label="new project name"
          value={this.state.name}
          type="text"
          placeholder="awesome project"
          onChange={this.dispatch}
        />
        <button onClick={this.addProject}>add</button>
        {projects && (
          <ProjectsContainer>
            {projects.map(project => (
              <li key={project.id}>
                <ProjectLink to={`/project/${project.id}`}>
                  {project.name}
                </ProjectLink>
              </li>
            ))}
          </ProjectsContainer>
        )}
      </SidebarContainer>,
    ];
  }
}

const PROJECTS_QUERY = gql`
  query projects {
    projects {
      id
      name
    }
  }
`;
const ADD_PROJECT_MUTATION = gql`
  mutation addProject($name: String!) {
    createProject(name: $name) {
      id
      name
    }
  }
`;

export default compose(
  graphql(PROJECTS_QUERY, { name: 'projectsQuery' }),
  graphql(ADD_PROJECT_MUTATION, { name: 'addProjectMutation' }),
)(Projects);
