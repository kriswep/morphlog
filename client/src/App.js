import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import Header from './component/Header';
import Home from './component/Home';
import Profile from './component/Profile';
import Projects from './component/Projects';
// import Project from './component/Project';

import theme from './styles/theme';
// import media from './styles/media';

// const Main = styled.section`
//   min-height: 100vh;
//   color: ${props => props.theme.darkShades};
//   background: ${props => props.theme.lightShades};
//   line-height: 1.625rem;
//   display: grid;
//   grid-gap: 3px;
//   grid-template-columns: 17.5rem 1fr;
//   grid-template-rows: auto 1fr auto auto;
//   grid-template-areas:
//     'header header' 'content content' 'sidebar sidebar' 'footer footer';
//   ${media.m`
//   grid-gap: 10px;
//   grid-template-rows: auto 1fr auto;
//   grid-template-areas: 'header header header' 'sidebar content content' 'footer footer footer';
//   `} ${media.l`
//     grid-template-columns: 20rem 1fr;
//   `} ${media.xl`
//     grid-template-columns: 35rem 1fr;
//   `} ${media.xxl`
//     grid-template-columns: 50rem 1fr;
//   `} ${media.xxxl`
//     grid-template-columns: 70rem 1fr;
//   `} ${media.uxxxl`
//     grid-template-columns: 100rem 1fr;
//   `};
// `;

class App extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <div>
          <Route path="/" component={Header} />
          <Route exact path="/" component={Home} />
          <Route path="/project/:projectId?" component={Projects} />
          <Route exact path="/profile" component={Profile} />
        </div>
      </ThemeProvider>
    );
  }
}

export default App;
