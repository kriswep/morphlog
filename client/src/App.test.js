/* globals test expect jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from './App';

// mockout these, since they produce errors
import Profile from './component/Profile';
import Projects from './component/Projects';
import Project from './component/Project';

jest.mock('./component/Profile', () => jest.fn());
jest.mock('./component/Projects', () => jest.fn());
jest.mock('./component/Project', () => jest.fn());

test('renders without crashing and correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toExist();
  expect(toJson(wrapper)).toMatchSnapshot();
});
