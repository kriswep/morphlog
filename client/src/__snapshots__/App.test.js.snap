// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "altPositiveColor": "#2b8e38",
      "altPositiveDarkColor": "#22732d",
      "darkAccent": "#454346",
      "darkShades": "#251D32",
      "darkestShades": "#09070c",
      "lightAccent": "#B8A896",
      "lightShades": "#F4F2F3",
      "lightestAccent": "#F1DFF0",
      "main": "#DC680B",
      "positiveColor": "#39b54a",
      "positiveDarkColor": "#34aa44",
    }
  }
>
  <div>
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/project/:projectId?"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/profile"
    />
  </div>
</ThemeProvider>
`;
